topics/lmms/rqtl2/gn-rqtl-design-implementation.gmi

```diff
@@ -0,0 +1,118 @@
+# RQTL Implementation for GeneNetwork Design Proposal
+
+## Tags
+
+* Assigned: alexm,
+* Keywords: RQTL, GeneNetwork2, Design
+* Type: Enhancements,
+* Status: In Progress
+
+
+
+## Description
+
+This document outlines the design proposal for the re-implementation of the RQTL feature in GeneNetwork providing also  a console view to track the external process.
+
+### Problem Definition
+The current RQTL implementation requires enhancements. The core functionality remains the same: making API calls from GeneNetwork 3 (GN3). The system needs a cleaner architecture for future use, emphasizing improved error handling and a clear separation of concerns between GeneNetwork 2 (GN2) and GN3. This will involve eliminating file transfers between GN2 and GN3. Additionally, a console should be provided to users for tracking the progress of ongoing tasks.
+
+## High-Level Design
+This is divided into two major components:
+* RQTL Api
+
+* Monitoring system for the rqtl script
+
+### RQTL Api 
+This component serves as the entry point to the gn3 rqtl system and can be subdivided into
+data validation and preprocessing stage, data computation stage, output processing stage.
```

Comment:
Can we have a breakdown of what each of these steps means in the context of RQTL, i.e.

```
- data validation: ensure the input files don't have error1, error2. error3 ..... etc.
- preprocessing: modify the input files by action1, action2, ... etc.
- data computation: ....
- output processing: ....
```
this would answer what the difference between output processing and data computation is for example.

Reply:
this commit addresses this issue: https://github.com/genenetwork/gn-gemtext-threads/pull/46/commits/9b591a22894c8ad37934e9cfb4b94cf9112f7067